sdk/client.py

```diff
@@ -146,21 +41,66 @@ def __init__(
         password: Optional[str] = None,
         token_provider: Optional[TokenProvider] = None,
         endpoints: Optional[Endpoints] = None,
+        auth0: Optional[Auth0Conf] = None,
     ):
-        if not (username and password) and not token_provider:
-            raise Exception(
-                "At least a username/password combination or"
-                "TokenProvider object should be provided."
+        if not username and not token_provider:
```

Comment:
This complex logics warrants a docstring which explains the different possibilities for user to instantiate the Client, with examples. Something like in the Readme imo (maybe we could even omit the token provider part in the readme and add it here only)

Reply:
I'm confused because the logic is the same as before, but yeah, I'll do it...

Reply:
```python
        """ In order to login, you have to provide either a login/password,
            or a token_provider.
            If you do not provide a password in order to fetch a token, that
            will be used to perform actions from this class, you will be prompted
            to enter one.
        """
```